hazelcast/src/main/java/com/hazelcast/topic/impl/reliable/ReliableTopicProxy.java

```diff
@@ -184,6 +194,22 @@ public void publish(@Nonnull E payload) {
         }
     }
 
+    @Override
+    public CompletionStage<Void> publishAsync(@Nonnull E message) {
+        checkNotNull(message, NULL_MESSAGE_IS_NOT_ALLOWED);
```

Comment:
Do we have to honor the `overloadPolicy` here? Seems like we do in `publish` and we also increment the stats (keep in mind that you need to check if we increment per-message or per-operation). Same comment for `publishAll` and `publishAllAsync`

Reply:
Ok, I've changed (and pushed) it but the class with all it methods looks like a big mess.
I will try to cleanup some things tomorrow.

Reply:
Ok, I've fixed and cleaned it up a little bit. The only thing actually missing is the dedicated test for `addAsyncAndBlock`.